### PR TITLE
sm7238: reverse video, 80/132 column switching, alternate font

### PR DESCRIPTION
**Pull request by shattered**

### CODE REVIEW

File: src/mame/drivers/sm7238.cpp [hunk 1]
```diff
@@ -189,28 +210,10 @@ WRITE_LINE_MEMBER(sm7238_state::write_printer_clock)
...
 void sm7238_state::recompute_parameters()
 {
-	rectangle visarea;
-	int horiz_pix_total = m_video.stride * 8;
-
-	visarea.set(0, horiz_pix_total - 1, 0, KSM_DISP_VERT - 1);
-	machine().first_screen()->configure(horiz_pix_total, KSM_DISP_VERT, visarea,
-		HZ_TO_ATTOSECONDS((m_video.stride == 80) ? 60 : 57.1 ));
+	machine().first_screen()->set_visible_area(m_video.stride, m_video.stride + (m_video.stride * 8) - 1,
+		KSM_VERT_START, KSM_VERT_START + KSM_DISP_VERT - 1);
```
Comment (cuavas) [inline src/mame/drivers/sm7238.cpp]: On the real machine, how does it implement the mode switch?  Does it change the pixel clock?  If so, does the line frequency change?  Or does it really just change the timing of horizontal blanking?

Comment (shattered) [reply src/mame/drivers/sm7238.cpp]: Manual says that pixel clock changes between 12.5 and 20.625 MHz.

Comment (cuavas) [reply src/mame/drivers/sm7238.cpp]: In that case, shouldn't it be calling `configure` as it was previously and updating timings?  Just changing visible area would only be correct if it's only changing blanking period.  Or do the ratios somehow work out so the line frequency is identical?

Comment (rb6502) [reply src/mame/drivers/sm7238.cpp]: Even if the ratios work out, I think it'd be preferable to use configure(), just for the documentation value on the changing pixel clock.

Comment (shattered) [reply src/mame/drivers/sm7238.cpp]: configure() isn't working for me, haven't found the reason yet.

**Comment by cuavas** [comment]
This should really find its screen by tag or something rather than always assuming first screen, but it's OK for now.

**Pull request merged**